File: output/gemma4-e4b/avatar/Python/Java/codeforces_242_A.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class codeforces_242_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read x, y, a, b from input
        if (!scanner.hasNextInt()) {
            // Handle case where input might be missing or malformed
            return;
        }
        
        int x = scanner.nextInt();
        int y = scanner.nextInt();
        int a = scanner.nextInt();
        int b = scanner.nextInt();
        
        List<int[]> games = new ArrayList<>();
        
        // Outer loop for i: range [a, x]
        for (int i = a; i <= x; i++) {
            // Condition: if (i <= b) continue
            if (i <= b) {
                continue;
            }
            
            // Inner loop for j: range [b, y]
            for (int j = b; j <= y; j++) {
                // Condition: if i > j
                if (i > j) {
                    // Store the pair (i, j)
                    games.add(new int[]{i, j});
                }
            }
        }
        
        // Print the length of games
        System.out.println(games.size());
        
        // Print each pair
        for (int[] pair : games) {
            System.out.println(pair[0] + " " + pair[1]);
        }
        
        scanner.close();
    }
}

File: output/qwen3.5-9b/avatar/Java/Python/codeforces_242_A.py
import sys

def main():
    x = int(input())
    y = int(input())
    a = int(input())
    b = int(input())
    
    count = 0
    for i in range(a, x + 1):
        for j in range(b, y + 1):
            if i <= j:
                continue
            count += 1
    
    print(count)
    
    for i in range(a, x + 1):
        for j in range(b, y + 1):
            if i <= j:
                continue
            print(i, j)

if __name__ == "__main__":
    main()